packages/material-ui/src/Box/Box.test.js

```diff
@@ -57,4 +58,98 @@ describe('<Box />', () => {
     expect(element.getAttribute('font-family')).to.equal(null);
     expect(element.getAttribute('font-size')).to.equal(null);
   });
+
+  it('respect properties order when generating the CSS', function test() {
+    const isMozilla = window.navigator.userAgent.indexOf('Firefox') > -1;
+    const isJSDOM = /jsdom/.test(window.navigator.userAgent);
+
+    if (isJSDOM || isMozilla) {
+      // Test fails on Mozilla with just:
+
+      // "border": "",
+      // "border-color": "",
+
+      this.skip();
+    }
+
+    const theme = createMuiTheme({
+      palette: {
+        primary: {
+          // light: will be calculated from palette.primary.main,
+          main: 'rgb(0, 0, 255)',
+          // dark: will be calculated from palette.primary.main,
+          // contrastText: will be calculated to contrast with palette.primary.main
+        },
+      },
+    });
+
+    const testCaseBorderColorWins = render(
+      <ThemeProvider theme={theme}>
+        <Box border={1} borderColor="primary.main" />
+      </ThemeProvider>,
+    );
+
+    expect(testCaseBorderColorWins.container.firstChild).toHaveComputedStyle({
+      border: '1px solid rgb(0, 0, 255)',
+      'border-color': 'rgb(0, 0, 255)',
```

Comment:
I wonder if it's not redundant, maybe only?

```suggestion
      border: '1px solid rgb(0, 0, 255)',
```

Reply:
True, this is enough